shared/src/test/scala/mlscript/DiffTests.scala

```diff
@@ -294,6 +294,10 @@ class DiffTests extends org.scalatest.funsuite.AnyFunSuite with org.scalatest.Pa
               ).foreach(td => tsTypegenCodeBuilder.declareTypeDef(td))
             }
 
+            // intialize pre-defined type defs with their variance
+            ctx.tyDefs.get("Array").foreach(tydef => tydef.tvarVariances += tydef.tparamsargs(0)._2 -> typer.VarianceInfo.co)
+            ctx.tyDefs.get("MutArray").foreach(tydef => tydef.tvarVariances += tydef.tparamsargs(0)._2 -> typer.VarianceInfo.in)
+
```

Comment:
This is really not the place to do it. It should be done in `Typer.scala`, in the same place we define the actual type defs:

https://github.com/hkust-taco/mlscript/blob/21ed4a4c886e5ce55d29849778bec509abe7a639/shared/src/main/scala/mlscript/Typer.scala#L119-L126